smash-servant-types/src/Cardano/SMASH/DBSync/Db/Types.hs

```diff
@@ -0,0 +1,145 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+
+module Cardano.SMASH.DBSync.Db.Types where
+
+import           Cardano.Prelude
+
+import           Control.Monad.Fail (fail)
+
+import           Data.Aeson (FromJSON (..), ToJSON (..), object, withObject, (.:), (.=))
+import           Database.Persist.Class
+
+import           Cardano.Api (AsType (..), Hash, deserialiseFromBech32, deserialiseFromRawBytesHex,
+                   serialiseToRawBytes)
+import           Cardano.Api.Shelley (StakePoolKey)
+import qualified Data.ByteString.Base16 as B16
+import qualified Data.ByteString.Char8 as BS
+
+import           Quiet (Quiet (..))
+
+-- | The stake pool identifier. It is the hash of the stake pool operator's
+-- vkey.
+--
+-- It may be rendered as hex or as bech32 using the @pool@ prefix.
+--
+newtype PoolId = PoolId { getPoolId :: Text }
```

Comment:
This should not be `Text`, it should be `ByteString` because we store the raw hashes in the database.

Reply:
Basically `cardano-db` should not be dependent on `smash-servant-types`.

Smash types in the database should be inline with other types in the database and not special Smash types.

Reply:
I already mentioned this. The wallet team depends on the servant types and they use them for their API, so they don't have to depend on the full `smash` repo. So while it might be possible to remove _some_ of the types, they would still need the `smash-servant-types` package.

Reply:
The data type presented to the wallet via `smash-servant-types` does not need to be the same as the type in the database.

Reply:
> The data type presented to the wallet via `smash-servant-types` does not need to be the same as the type in the database.

Can you clarify what you mean by that?

Reply:
The `cardano-db` package should not depend on `smash-servant-types`. The smash extensions to the database schema should be in terms of existing types in `cardano-db`. When smash presents data to the wallet, it can wrap them in whatever newtypes or whatever makes sense for the wallet.

Reply:
> The `cardano-db` package should not depend on `smash-servant-types`. The smash extensions to the database schema should be in terms of existing types in `cardano-db`. When smash presents data to the wallet, it can wrap them in whatever newtypes or whatever makes sense for the wallet.

To make this clear, you mean - remove `smash-servant-types` and put all the types in the `cardano-db` repo?

Reply:
No, I mean the smash db extensions should use `ByteString` for things like hashes, `Text` for human readable strings etc. When presenting these values to the wallet the can be converted to the types the wallet expects.

Reply:
So then I leave `smash-servant-types` as is and switch from `Text` to `ByteString`?